src/forms/medication-dispense-review.component.tsx

```diff
@@ -305,11 +305,7 @@ const MedicationDispenseReview: React.FC<MedicationDispenseReviewProps> = ({
           }
           min={0}
           max={config.dispenseBehavior.restrictTotalQuantityDispensed ? quantityRemaining : undefined}
-          value={
-            quantityRemaining <= medicationDispense.quantity.value
-              ? quantityRemaining
-              : medicationDispense.quantity.value
-          }
+          value={medicationDispense.quantity.value}
```

Comment:
So I want to revert this part of the previous commit for O3-3557, as it is restricting the quantity dispensed even if the quantity dispensed flag is not set.  I also think it's confusing, because it's not ever letting the user type in a number greater than the quantity remaining... I think it's better to allow the user to do this but give an error.  Also, as it currently works the "Dispense" button doesn't appear to be enabled until you re-type the number.

I think I do understand the reasoning for this now... prior to this change, when starting a new dispense, the default value on this form is set to whatever the order quantity was... which is likely *not* what you want if the order quantity is less than the quantity remaining.  However, if we want to enforce this, I think we need to do it when the the medication dispense object is initially set up (which I think is in the "initiateMedicationDispenseBody" method in the medication dispense resource).

Reply:
Note that admittedly I haven't tested this with the stock management functionality, as I don't have a demo environment for that, not sure if further changes are needed if we revert this to meet your use case.

Reply:
So Antony, from your views I think we can keep the filtering of the batches and on selecting that do not change the quantity to dispense ... Let that component remain the way it is.... So generally it will be a filter of batches but let the user input the dispense value

Reply:
Thanks, @FelixKiprotich350. Please go ahead and provide your review and approval.

Reply:
Yes @FelixKiprotich350  please let me know if this looks okay to you.  

Sorry if I don't understand the how the stock integration functionality it supposed to work.  If the idea is that selecting a batch should change the quantity to dispense, that seems doable, but I would think it should be done via function that is passed into the StockDispense component, like the setAvailableQuantity currently is.  I don't think we can just set the available quantity when the user selected a batch and then determine the dispense amount from that.   Does that make sense?  I'm happy to look at some screen shots to understand the stock dispense functionality better.  fyi @ojwanganto

Reply:
yeah sure this looks fine to let the user provide his quantity

Reply:
hello @mogoodrich i am attaching a short video of how the component works within the stock-dispensing  module
[screen-capture.webm](https://github.com/user-attachments/assets/44fef431-127e-4504-a93c-e50231d82dba)